client/components/week-progress-chart/week-chart.jsx

```diff
@@ -0,0 +1,134 @@
+    this.setState({
+      SVGwidth: DIVref.current.offsetWidth,
+      chartWidth: DIVref.current.offsetWidth - 25
+    }, () => drawChart());
+    window.addEventListener('resize', resizeChart);
+  }
+
+  drawChart() {
+    const { chartHeight, height, marginLeft, marginTop } = this.dimensions;
+    const { SVGwidth, chartWidth } = this.state;
+    const { DIVref } = this;
+    const { data } = this.props;
+
+    const chartSVG = select(DIVref.current)
+      .append('svg')
+      .attr('width', SVGwidth)
+      .attr('height', height)
+      .classed('svg-chart', true);
+
+    const x = scaleBand()
+      .domain(data.map(d => d.date))
+      .range([0, chartWidth - 0.5 * marginLeft]);
+
+    const y = scaleLinear()
+      .domain([0, Math.floor((max(data, d => d.distance) + 1))])
+      .range([chartHeight, 0]);
+
+    const xAxis = axisBottom(x)
+      .tickSizeOuter(0);
+
+    const yAxis = axisLeft(y)
+      .tickSizeOuter(0)
+      .ticks(10)
+      .tickFormat((interval, i) => {
+        return i % 2 !== 0 ? '' : `${interval}mi`;
+      });
+
+    // Axis //
+    const xAxisGroup = chartSVG
+      .append('g')
+      .attr('transform', `translate(${marginLeft}, ${chartHeight + marginTop})`)
+      .call(xAxis);
+
+    xAxisGroup
+      .selectAll('text')
+      .attr('text-anchor', 'center')
+      .classed('text-[10px] xxs:text-[12px]', true);
+
+    const yAxisGroup = chartSVG
+      .append('g')
+      .attr('transform', `translate(${marginLeft}, ${marginTop})`)
+      .call(yAxis);
+
+    yAxisGroup
+      .selectAll('text')
+      .attr('text-anchor', 'center')
+      .attr('font-size', '11px');
+
+    // Line //
+    chartSVG
+      .append('path')
+      .data([data])
+      .attr('class', 'line')
+      .attr('fill', 'none')
+      .attr('transform', `translate(${marginLeft + (chartWidth * 0.070)}, ${marginTop})`)
+      .attr('stroke-width', 2.5)
+      .attr('stroke', 'steelblue')
+      .attr('d', line()
+        .x(d => { return x(d.date); })
+        .y(d => { return y(d.distance); })
+      );
+
+    // Data Points //
+    chartSVG
+      .selectAll('circle')
+      .data(data)
+      .enter()
+      .append('circle')
+      .attr('transform', `translate(${marginLeft + (chartWidth * 0.070)}, ${marginTop})`) // transforms entire graph
+      .attr('fill', 'lightblue')
+      .attr('stroke', 'darkblue')
+      .attr('stroke-width', 1.5)
+      .attr('cx', function (d) { return x(d.date); })
+      .attr('cy', function (d) { return y(d.distance); })
+      .attr('r', 3.5);
+  }
+
+  resizeChart() {
+    const { DIVref, drawChart } = this;
+    if (!DIVref || !DIVref.current) {
+      return;
+    }
+    const currentChart = select('.svg-chart');
+    currentChart.remove();
+    this.setState({
+      SVGwidth: DIVref.current.offsetWidth,
+      chartWidth: DIVref.current.offsetWidth - 25
+    }, () => drawChart());
+  }
+
+  render() {
+    if (!this.props) {
+      return;
+    }
```

Comment:
`this.props` is always guaranteed to at least be an empty object, so this guard isn't doing anything for you.